internal/brokers/broker_test.go

```diff
@@ -0,0 +1,300 @@
+package brokers_test
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/internal/brokers"
+	"github.com/ubuntu/authd/internal/responses"
+	"github.com/ubuntu/authd/internal/testutils"
+)
+
+func TestNewBroker(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		name       string
+		configFile string
+		wantErr    bool
+	}{
+		"No config means local broker":                        {name: "local"},
+		"Successfully create broker with correct config file": {name: "broker", configFile: "Broker"},
+
+		"Error when config file is invalid": {configFile: "invalid", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			conn, err := testutils.GetSystemBusConnection()
+			require.NoError(t, err, "Setup: could not connect to system bus")
+
+			if tc.configFile != "" {
+				tc.configFile = filepath.Join(brokerDir, tc.configFile)
+			}
+
+			got, err := brokers.NewBroker(context.Background(), tc.name, tc.configFile, conn)
+			if tc.wantErr {
+				require.Error(t, err, "NewBroker should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "NewBroker should not return an error, but did")
+
+			gotString := fmt.Sprintf("ID: %s\nName: %s\nBrand Icon: %s\n", got.ID, got.Name, got.BrandIconPath)
+
+			wantString := testutils.LoadWithUpdateFromGolden(t, gotString)
+			require.Equal(t, wantString, gotString, "NewBroker should return the expected broker, but did not")
+		})
+	}
+}
+
+func TestNewSession(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		username string
+
+		wantErr bool
+	}{
+		"Successfully start a new session": {username: "test"},
+
+		// broker errors
+		"Error when starting a new session": {username: "NS_error", wantErr: true},
+
+		// dbus errors
+		"Error when calling bus method": {username: "NS_bus_error", wantErr: true},
```

Comment:
Those 2 tests are basically the same from a brokers package perspective. so, they don’t add much and merging the objects will help (ditto on the other similar tests)